coordinator/src/main/java/org/apache/uniffle/coordinator/strategy/storage/AppBalanceSelectStorageStrategy.java

```diff
@@ -84,23 +85,35 @@ public void sortPathByRankValue(String path, String test) {
   public void detectStorage() {
     uris = Lists.newCopyOnWriteArrayList(remoteStoragePathRankValue.entrySet());
     if (remoteStoragePathRankValue.size() > 1) {
+      CountDownLatch countDownLatch = new CountDownLatch(uris.size());
       for (Map.Entry<String, RankValue> uri : uris) {
-        if (uri.getKey().startsWith(ApplicationManager.getPathSchema().get(0))) {
-          RankValue rankValue = remoteStoragePathRankValue.get(uri.getKey());
-          rankValue.setHealthy(new AtomicBoolean(true));
-          Path remotePath = new Path(uri.getKey());
-          String rssTest = uri.getKey() + "/rssTest-" + getCoordinatorId();
-          Path testPath = new Path(rssTest);
-          try {
-            FileSystem fs = HadoopFilesystemProvider.getFilesystem(remotePath, hdfsConf);
-            readAndWriteHdfsStorage(fs, testPath, uri.getKey(), rankValue);
-          } catch (Exception e) {
-            rankValue.setHealthy(new AtomicBoolean(false));
-            LOG.error("Storage read and write error, we will not use this remote path {}.", uri, e);
-          } finally {
-            sortPathByRankValue(uri.getKey(), rssTest);
+        Thread detectThread = new Thread(() -> {
```

Comment:
I mean use `uris.stream().parallel()`

Reply:
I use `parallelStream`, the two should be similar. right?

Reply:
Yes